File: backend/src/main/java/com/runway/run/dto/RunningStatsResponse.java
package com.runway.run.dto;

import lombok.Builder;
import lombok.Getter;

import java.time.Instant;

@Getter
@Builder
public class RunningStatsResponse {

    private String period;
    private long totalRuns;
    private double totalDistanceMeters;
    private long totalDurationSeconds;
    private long totalCaloriesBurned;
    private int averagePaceSecondsPerKm;
    private double longestRunMeters;
    private int currentStreakDays;
    private int longestStreakDays;
    private long activeDays;
    private Instant periodStart;
    private Instant periodEnd;
}

File: android/app/src/main/java/com/runway/android/data/running/model/RunningStatsResponse.kt
package com.runway.android.data.running.model

data class RunningStatsResponse(
    val period: String,
    val totalRuns: Long,
    val totalDistanceMeters: Double,
    val totalDurationSeconds: Long,
    val totalCaloriesBurned: Long,
    val averagePaceSecondsPerKm: Int,
    val longestRunMeters: Double,
    val currentStreakDays: Int,
    val longestStreakDays: Int,
    val activeDays: Long,
    val periodStart: String?,
    val periodEnd: String?,
)
